src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -43,6 +43,9 @@ Copyright (c) .NET Foundation. All rights reserved.
     or TargetFrameworkVersion that is already set.
    -->
 
+  <!-- Hook for importing custom target framework parsing -->
+  <Import Project="$(CustomTargetFrameworkParsingTargets)" Condition="$(CustomTargetFrameworkParsingTargets) != '' and Exists($(CustomTargetFrameworkParsingTargets))" />
```

Comment:
I think we might want to remove the Exists check.  If you've set the property to a nonexistent file, you probably want it to error instead of silently ignoring the property.

Reply:
Note the comment at the top of this file:

> Note that this file is only included when $(TargetFramework) is set and so we do not need to check that here.

Is it OK to only import the custom TargetFramework parsing .targets when `TargetFramework` is set?  Or might there be scenarios where `TargetFramework` isn't set at all and you want your custom logic to set it?  @Anipik @ericstj

Reply:
`CustomTargetFrameworkParsingTargets` seems like a pretty good name for this property, but does anyone have any other suggestions?  @rainersigwald @wli3 @Anipik @ericstj

Reply:
I think its ok to only do this with `TargetFramework` is set.  @Anipik?

Do we use the term `Custom` elsewhere?  That seems a bit unusual to me, but if we have that pattern elsewhere its ok.  Another ideas for a name would be to reuse the `TargetFrameworkInference` term, like `BeforeTargetFrameworkInferenceTargets` or something.

Reply:
We do have properties such as `CustomBeforeMicrosoftCommonProps`.

However, I do think `BeforeTargetFrameworkInferenceTargets` or `CustomBeforeTargetFrameworkInferenceTargets` would be better for this property name.

Reply:
Those targets have that name pattern because the `Microsoft.*.targets` are public. This target is internal. So, an user-friendly name (that describes the intent) would be better, IMHO!

Reply:
This is meant to be a public extension point.  It's mainly intended for the dotnet/runtime repo to use, but others could.

I still lean somewhat towards the names that match the existing `BeforeXYZTargets` pattern.

Reply:
`BeforeTargetFrameworkInferenceTargets` sounds like a good name to me, I can update to that.

Reply:
What I meant, is the `TargetFrameworkInference` targets is internal!

Reply:
> I think its ok to only do this with TargetFramework is set. @Anipik?

Yes its ok to import this when target framework is set